### ISSUE TITLE
Interface with contacts and share menu

### ISSUE THREAD

**Issue by S7venLights** [TrianguloY/OpenInWhatsapp_Web#1]
I was looking for Click to chats repo and this is the closest I could find. 
I was wondering why click to chat doesn't work via the share intent or use the contact permission to aid quick opening of chats by selecting a contact?
The old open in WhatsApp app used to do this.

An alternative would be for the app to act as a kind of contacts app where you can save peoples contact and just search and click on them from 'Click to Chat' to open in WhatsApp, instead of having to copy paste from different apps.

**Comment by S7venLights** [comment]
Actually that just made me think I can create a
https://wa.me/<Number>?text=<message+Text> 
Link saved directly in my contacts app to open in WhatsApp. Much easier.

**Comment by TrianguloY** [comment]
This repo is for the web version, are you referring to it or to the [Android version](https://github.com/TrianguloY/OpenInWhatsapp_Translation)?

In any case, the app does not use the contacts permission because it is designed to work for numbers not in your contacts (otherwise the app is useless, just open WhatsApp). 
It does work for the share intent though.
And also, there is a list of recently opened numbers that you can pin to keep some sort of contacts list.

<br>

> The old open in WhatsApp app used to do this.

My app has never have the contacts permission, and no updates since it was renamed should have broken the share functionality. Maybe you switched device? The newer android versions have a lot of restrictions to how apps can interact with some, maybe there is an issue with one of those? In that case I recommend to [create a new issue in the other repository](https://github.com/TrianguloY/OpenInWhatsapp_Translation/issues/new). 
<sub>Or maybe you are thinking to one of those other clones of my app? If it had ads or shady permissions it wasn't mine.</sub>

**Comment by S7venLights** [comment]
I was thinking of this app: [Link removed]

"In any case, the app does not use the contacts permission because it is designed to work for numbers not in your contacts (otherwise the app is useless, just open WhatsApp)."
>My use case for this would be avoiding giving WhatsApp permission to access all my contacts for some extra privacy.

In any case my idea of saving the link in contacts works fine.

**Comment by S7venLights** [comment]
Thanks though (ツ)

**Comment by TrianguloY** [comment]
Yep, one of the first clones, even contains part of my assets and code (which is not public so it was obtained by decompiling without my permission and I could probably report it, but...to be honest I simply learned to just don't care. I removed the link though).

Anyway, as for the functionality, I'll check if the app can handle contact links without the contact permission, if so I may add it.
